yellowbrick/classifier.py

```diff
@@ -50,6 +51,149 @@ def __init__(self, model, ax=None, **kwargs):
+            indicated by that function. It may be used to reorder or select a subset of labels.
+            If None, values that appear at least once in y_true or y_pred are used in sorted order.
+            Default: None
+        """
+        super(ConfusionMatrix, self).__init__(model, ax=ax, **kwargs)
+        #Parameters provided by super (for reference during development only):
+        #self.ax
+        #self.size
+        #self.color
+        #self.title
+        #self.estimator
+        #self.name
+
+        #Initialize all the other attributes we'll use (for coder clarity)
+        self.confusion_matrix = None
+
+        self.cmap = color_sequence(kwargs.pop('cmap', 'YlOrRd'))
+
+        #If possible, assign the classes. If this can't happen now, it will happen in the .fit() method
+        #TODO should consider implementing this approach in ClassificationReport - allows the user to use
+            #the default classes_ list without calling fit (i.e. if the model has already been fit before
+            #initializing the visualizer)
+        if classes is None:
+            try:
+                self.classes_ = self.estimator.classes_
+            except AttributeError:
+                self.classes_ = None
+        else:
+            self.classes_ = classes
+
+    #TODO this is the same as ClassificationReport, should hoist it up a level
+    #TODO is there any way to get this documentation to extend the parent class documentation so we don't have to double(triple, quadruple...) maintain it?
+    def fit(self, X, y=None, **kwargs):
+        """
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs: keyword arguments passed to Scikit-Learn API.
+        """
+        super(ConfusionMatrix, self).fit(X, y, **kwargs)
+        if self.classes_ is None:
+            self.classes_ = self.estimator.classes_
+        return self
+
+    def score(self, X, y, sample_weight=None):
+        """
+        Generates the Scikit-Learn confusion_matrix and applies this to the appropriate axis
+
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        sample_weight: optional, passed to the confusion_matrix
+        """
+        y_pred = self.predict(X)
+        self.confusion_matrix = confusion_matrix(y_true = y, y_pred = y_pred, labels=self.classes_, sample_weight=sample_weight)
+        #Convert confusion matrix to percent of true
+        self.confusion_matrix = np.round(self.confusion_matrix / np.sum(self.confusion_matrix, axis=1) * 100, decimals=0)
```

Comment:
The issue we discussed is what to do when the raw numbers have too many digits to display properly for the size of the matrix. Options include:
- Display it anyways, formatting will break, and user will manually change to % mode. Kind of ugly.
- Use logic to decide to override the user choice and display it as percent when it's too large. (maybe raise a warning? Potentially confusing behavior that doesn't fit to user expectations)
- Only display the cells with too many digits as percent (potentially confusing when there are mixed meanings of numbers. Could use smart formatting to clarify) if whole number mode is selected. 
- Resize the font of all cells based on the max number of digits and available space. I would like to do this anyways to deal with situations that have lots of classes (like my example), but I need to find out how to get the display size of the graphic. This might mean adding the text needs to get moved into poof()? Do you know how to get pixel dimensions of the resulting graphic?
- Resize the font of each cell individually based on digit count. Same implementation issue as above bullet. 

Due to these complications, we thought we'd start first with only allowing the percent-based display format. What do you think about the choices between these options, for future implementation?